[MRG] Add depth units (µm) to LFP plot and reduce y-axis tick clutter

Shivansh1205 opened the pull request:
This PR adds micron units (µm) to the depth axis in the laminar LFP plot and reduces y-axis clutter by showing only every third contact label. This improves readability and ensures scientifically accurate depth labeling.

Shivansh1205 committed:
Add depth units (µm) to LFP plot and reduce y-axis tick clutter

ntolley commented:
Thanks for the PR @Shivansh1205 !

Do you mind posting a picture of the plot before/after this change for documentation purposes?

Shivansh1205 committed:
added a more compact version

your suggested code only generates labels for every 3rd value but it also reduces the number of tick labels, which Matplotlib will still match to all ticks. So I have a tweaked a few more changes.

Shivansh1205 committed:
Remove compiled NEURON mechanism files (.c/.o/.dll) and update .gitignore

Shivansh1205 committed:
Fix laminar LFP y-axis labeling: show all contact labels

Shivansh1205 committed:
Fix laminar LFP y-axis labeling: show all contact labels and remove depth units

Shivansh1205 commented:
Hi @ntolley,
- As requested, here is a summary of the changes:
- Removed depth units (µm) from the y-axis label in the laminar LFP plot.
- Restored visibility of all contact labels to improve interpretability.
- Cleaned up tick labeling logic for consistency.
- Verified that unit tests pass successfully after the update.
- Formatting and styling issues (ruff) have been fixed.
Let me know if you would like before/after plot visualizations, or any further refinements.
Thanks for the guidance!
— Shivansh

asoplata committed:
Revert deletion of LFP units and label skip

This reverts the deletions of bc98ad16 such that now, once again, the
y-axis labels are only displayed every 3 ticks (helps readability), and
the scientific units are displayed once again on the LFP y-axis
title.

I'm not sure why these were deleted, it may have been from
private communication between Nick and Shivansh, or it may have been a
miscommunication since nothing in the PR that I see by Nick gives
instructions to remove these valuable enhancements.

asoplata committed:
Remove MOD ignores from gitignore

These ignores were added unnecessarily, since compiled files should only
be output into either `hnn_core/mod/x86_64/` or `hnn_core/mod/arm64/`,
which are already included in gitignore.

asoplata committed:
Merge branch 'master' into feature/lfp-depth-units

asoplata committed:
DOC add contribution

asoplata commented:
I've added some new commits to Shivansh's branch that re-apply some of the y-axis changes which I believe were removed in error. I've also cleaned up the gitignore and have brought the branch up to date with `master`.

This is now ready for merge to master, pending CI.

asoplata commented:
As an example of how much better it looks, here is the updated output fig from running https://github.com/jonescompneurolab/hnn-core/blob/master/examples/howto/plot_record_extracellular_potentials.py

<img width="600" height="800" alt="lfp-csd" src="https://github.com/user-attachments/assets/f610c5fd-39bb-49fd-a4c3-05af51cbe496" />
